include/boost/serialization/throw_exception.hpp

```diff
@@ -18,7 +18,7 @@
 #include <boost/config.hpp>
 
 #ifndef BOOST_NO_EXCEPTIONS
```

Comment:
This needs to be an `#ifdef`, not an `#ifndef`.  Or just include it unconditionally since it contains a similar check internally.

Reply:
You're right, but that still doesn't look right to me somehow :(